File: templates/typescript/product.ts
interface Product {
  id: number;
  name: string;
  price: string;
  description: string;
}

fetch('http://localhost:3001/api/product')
  .then(response => response.json())
  .then((data: Product) => console.log('Product:', data))
  .catch((error: any) => console.error('Error:', error));

File: templates/csharp/Product.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

class Program {
  static async Task Main() {
    using var client = new HttpClient();
    try {
      var response = await client.GetStringAsync("http://localhost:3001/api/product");
      Console.WriteLine("Product: " + response);
    } catch (Exception e) {
      Console.WriteLine("Error: " + e.Message);
    }
  }
}